modules/artifact/src/main/java/org/mule/module/artifact/classloader/ThreadCommonMethodsUtil.java

```diff
@@ -0,0 +1,104 @@
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.artifact.classloader;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+import org.mule.runtime.module.artifact.api.classloader.ArtifactClassLoader;
+import org.mule.runtime.module.artifact.api.classloader.MuleArtifactClassLoader;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.List;
+import org.slf4j.Logger;
+
+
+/**
+ * This class contain common methods used by ActiveMQResourceReleaser and JdbcResourceReleaser Used to prevent leaks related to
+ * threads
+ */
+public class ThreadCommonMethodsUtil {
+
+  private static final Logger logger = getLogger(ThreadCommonMethodsUtil.class);
+  public static final String COMPOSITE_CLASS_LOADER_CLASS_NAME = "CompositeClassLoader";
+
+  public static boolean isThreadLoadedByDisposedDomain(String undeployedArtifactId, ClassLoader threadContextClassLoader) {
+    try {
+
+      Class threadContextClassLoaderClass = threadContextClassLoader.getClass();
+      if (!threadContextClassLoaderClass.getSimpleName().equals(COMPOSITE_CLASS_LOADER_CLASS_NAME)) {
+        return false;
+      }
+
+
+      Method getDelegateClassLoadersMethod = threadContextClassLoaderClass.getMethod("getDelegates");
+      List<ClassLoader> classLoaderList = (List<ClassLoader>) getDelegateClassLoadersMethod.invoke(threadContextClassLoader);
+
+
+      for (ClassLoader classLoaderDelegate : classLoaderList) {
+        if (classLoaderDelegate instanceof ArtifactClassLoader) {
+          ArtifactClassLoader artifactClassLoader = (ArtifactClassLoader) classLoaderDelegate;
+          if (artifactClassLoader.getArtifactId().contains(undeployedArtifactId)) {
+            return true;
+          }
+        }
+      }
+
+    } catch (Exception e) {
+      logger.warn("Exception occurred while attempting to compare {} and {} artifactId.", threadContextClassLoader,
+                  threadContextClassLoader.getClass().getClassLoader());
+    }
+    return false;
+  }
+
+  public static boolean isThreadLoadedByDisposedApplication(String undeployedArtifactId, ClassLoader threadContextClassLoader) {
+    try {
+      if (!(threadContextClassLoader instanceof MuleArtifactClassLoader)) {
+        return false;
+      }
+      String threadClassLoaderArtifactId = ((MuleArtifactClassLoader) threadContextClassLoader).getArtifactId();
+      return threadClassLoaderArtifactId != null && threadClassLoaderArtifactId.equals(undeployedArtifactId);
+    } catch (Exception e) {
+      logger.warn("Exception occurred while attempting to compare {} and {} artifact id.", threadContextClassLoader,
+                  threadContextClassLoader.getClass().getClassLoader());
+    }
+    return false;
+  }
+
+  public static void clearReferencesStopTimerThread(Thread thread)
```

Comment:
this seems to be soecific for a certain driver, not general to be placed in this utils class

Reply:
But this code works fine also with ActiveMQ driver, and is used in both classes. What you propose?

Reply:
Yes, Right. java.util.TimerThread is a class Is used for both classes.